### ISSUE TITLE
extract_instruments problem

### ISSUE THREAD

**Issue by jonnyrose** [MRCIEU/TwoSampleMR#68]
When I go to MR Base (and try to run an MR with UKB-b:5779 as exposure), I get a message that none of the requested outcomes had GWAS hits.  When I use two-sample MR to extract instruments for alcohol intake frequency (UKB-b:5779), I get the following same message:

Requesting default values. Extracting from pre-clumped data
None of the requested outcomes had GWAS hits at the specified threshold.
Warning message:
In extract_instruments(outcomes = "UKB-b:5779") :
  From version 0.4.2 the exposure name format has changed.

So that didn't make sense.  So I tried extract_instruments, clump=FALSE, and I get 8407 snps (all with pvalue < 5e-08).  Then when I clump using clump_data, I get 99 snps.  

I looked at alcohol intake frequency from the Neale Lab (GWAS library UKB-a:25).  I get 44 snps whether I extract_instruments with clump=TRUE or extract_instruments (clump=FALSE) then clump_data.  

Then I tried something else:
>alc5779_c <-extract_instruments(outcomes = "UKB-b:5779", p1=5e-07, clump=TRUE, p2=1,r2=.001, kb=10000)
Extracting data from 1 GWAS(s)
and performing LD clumping
 [>] 1 of 1
Token cache file: mrbase.oauth
Warning messages:
1: In extract_instruments(outcomes = "UKB-b:5779", p1 = 5e-07, clump = TRUE,  :
  From version 0.4.2 the phenotype format has changed.
2: In format_data(d, type = "exposure", snps = NULL, phenotype_col = "phenotype",  :
  n column is not numeric
> alc5779_c08 <- subset(alc5779_c, alc5779_c$pval.exposure <.00000005)
Which gave me 99 snps.

(I also tried both p2=5e-08 and p2=5e-07 with p1=5e-07 and I got 157 down to 99 snps when I subsetted for pval.exposure < .00000005)

Just asking, because I was thrown off by MR Base returning that there were no instruments for the exposure.

Thanks.

**Comment by mightyphil2000** [comment]
Thanks for letting us know. the datasets have been preclumped using a threshold for p1 of 5e-8. The default is to search for SNPs in this preclumped dataset (ie clump=FALSE). However, i suspect that the alcohol dataset, which is relatively new, is missing from this preclumped set. This would explain why it works if clump=TRUE and doesn't work when set to FALSE. If I'm right, we will have to update the set of preclumped datasets or modify the code to override the default behaviour when the exposure dataset has not already been preclumped.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by mightyphil2000** [comment]
Do you only see this problem for datasets with ID beginning in "UKB-b:..."?

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by ValeriiaH** [comment]
There is a relatively new option "force_server" in extract_instruments() which I believe does what you want. It will force the instruments to be extracted directly from the server rather than from the preclumped repository. Hence, you may wish to try something like extract_instruments(<IDs>,force_server=TRUE). It is set to be FALSE by default.

**Comment by mightyphil2000** [comment]
oh yah thanks Val. Good suggestion. Jonnyrose does it work if you do that?


Sorry I made some mistakes in my previous message. The default is force_server=FALSE, not clump=FALSE.


If the data has not been preclumped already that might explain why it doesn't work when force_server=FALSE, clump=TRUE and p1<5e-8. the preclumped repository is based on p1<5e-8. If p1 is different from this, then the script is forced to extract from the server, hence why it works in that circumstance. As Val said, you can get around this problem by setting force_server to TRUE.


bw

P



________________________________

**Comment by jonnyrose** [comment]
Yes thanks!  Problem was only for the alcohol phenotypes from MRC-IEU (UKB-b:).  No problem with other phenotypes that I tried.  Force_server worked as did my ham-fisted work-arounds (clump=FALSE, then clump_data).

**Comment by jonnyrose** [comment]
Nope, sorry, spoke too soon.  Problem also for another phenotype, not a UKB-b:  Clarke et al GWAS on alcohol consumption, recently uploaded to GWAS library, id=1283.  4 instruments but only when I use force_server = TRUE.  Otherwise same "none of the requested outcomes. . . "  

Thanks again.

**Comment by ChristopherZouNMU** [comment]
Nice to meet you! I meet with a rare question. When I try to run an MR with UKB-b:XXX as exposure, it shows lexical error: invalid char in json text. <!DOCTYPE HTML PUBLIC "-//W3C//. Can you help me with this? Other people seem to have never met this problem.
